src/platform/windows/misc.cpp

```diff
@@ -200,6 +202,71 @@ namespace platf {
     return std::string(buffer, bytes);
   }
 
+  /**
+ * @brief A function to duplicate the current sessions user's token with elevated privileges
+ *
+ * @return A handle to the duplicated users token, or null if the duplication failed
+ */
+  HANDLE
+  duplicate_users_token_elevated() {
+    DWORD consoleSessionId;
+    HANDLE userToken, duplicateToken;
+    TOKEN_ELEVATION_TYPE elevationType;
+    DWORD dwSize;
+
+    // Close the userToken handle when it goes out of scope
+    auto token_close = util::fail_guard([&userToken]() {
+      CloseHandle(userToken);
```

Comment:
This won't be initialized prior to a successful call to `WTSQueryUserToken()`, so you'll need to move this below that call.